crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -496,11 +530,30 @@ impl<'src> Parser<'src> {
         let start = self.node_start();
         self.bump(TokenKind::Nonlocal);
 
+        // test_err nonlocal_stmt_trailing_comma
+        // nonlocal ,
+        // nonlocal x,
+        // nonlocal x, y,
+
+        // test_err nonlocal_stmt_expression
+        // nonlocal x + 1
         let names = self.parse_comma_separated_list_into_vec(
             RecoveryContextKind::Identifiers,
             Parser::parse_identifier,
         );
 
+        if names.is_empty() {
+            // test_err nonlocal_stmt_empty
```

Comment:
Have you been doing this kind of thing elsewhere? Just seems hard to keep in-sync. If it's already an established pattern, no worries. Otherwise, I probably would suggest just documenting the cases, but not the test name, like:

```
// For example:
// ```python
// nonlocal
// ```
```

Reply:
These are inline tests which gets extracted automatically into their own test file and gets tested. 

Refer #10681 and [documentation](https://github.com/astral-sh/ruff/blob/dhruv/parser/crates/ruff_python_parser/CONTRIBUTING.md#inline-tests).

Reply:
That’s really cool!